web/modules/custom/lit_cover_service/src/Service/BatchService.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+namespace Drupal\lit_cover_service\Service;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\file\Entity\File;
+use Drupal\node\Entity\Node;
+
+/**
+ * Class BatchService.
+ */
+class BatchService {
+
+  /**
+   * Batch delete process callback.
+   *
+   * @param int $batchId
+   *   Id of the batch
+   * @param $batchTotal
+   *   Total number of batch operations
+   * @param array $nids
+   *   The nids to process
+   * @param object $context
+   *   Context for operations
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function deleteBookCovers($batchId, $batchTotal, $nids, &$context) {
+    if (!isset($context['sandbox']['progress'])) {
+      $context['sandbox']['progress'] = 0;
+      $context['sandbox']['current_node'] = 0;
+      $context['sandbox']['max'] = $batchTotal;
+    }
+
+    $context['message'] = t('Deleting batch @batchId of @batchTotal', [
+      '@batchId' => $batchId,
+      '@batchTotal' => $batchTotal,
+    ]);
+
+    $nodes = Node::loadMultiple($nids);
+    $deletedImageFiles = [];
+    foreach ($nodes as $node) {
+      $deletedImageFiles[] = self::clearImageField($node);
```

Comment:
`clearImageField` set the field to `null` will this actually delete the local file? I known that you call `deleteFileEntities` later on, but what if the first 50 nodes are cleared and the 51 fails with an exception?

Reply:
Tradeof between  efficiency and complexity of the error handling. 
We can probably move the file deletion so we do it immediately to avoid this without it being to much slower.